internal/action/main.go

```diff
@@ -7,49 +7,74 @@ import (
 	"io/ioutil"
 	"os"
 	"path"
-	"strings"
 
 	"github.com/hashicorp/go-tfe"
 	"github.com/hashicorp/terraform-exec/tfexec"
 	"github.com/sethvargo/go-githubactions"
-	"github.com/takescoop/terraform-cloud-workspace-action/internal/action/inputs"
 	"github.com/takescoop/terraform-cloud-workspace-action/internal/tfconfig"
 	"github.com/takescoop/terraform-cloud-workspace-action/internal/tfeprovider"
 	yaml "gopkg.in/yaml.v2"
 )
 
-func Run() {
-	ctx := context.Background()
+type RunConfig struct {
+	Token                  string
+	Host                   string
+	Name                   string
+	Organization           string
+	Apply                  bool
+	RunnerTerraformVersion string
+	RemoteStates           string
+	Workspaces             string
+	Variables              string
+	WorkspaceVariables     string
+	TeamAccess             string
+	BackendConfig          string
+	AgentPoolID            string
+	AutoApply              *bool
```

Comment:
Seems like these pointers can be eliminated. Thought about this working on https://github.com/bendrucker/go-githubactions-inputs. Inputs either have defaults or are required. There is no such thing as a nil action input.

Reply:
I'm fine with that, what do you think about adopting go-githubactions-inputs and the no nil action inputs in a separate upcoming PR? 

As you've likely considered, this would mean giving up the ability to fall through to defaults on the Terraform provider. Not a bad thing really, but we'd want to match those defaults I think

Reply:
That's fine, and now that I actually look I'm reminded that `""` equates to nil as implemented in the package in this repo. Will keep thinking on this a bit. Part of the motivation is that allocating pointers with reflection is a much bigger pain than setting values. But ultimately if there's a sensible use I'll just suck it up and handle them.